File: fixtures/scale/csharp/small/Render/Alignment.cs
using TextPrint.Util;

namespace TextPrint.Render
{
    public static class Alignment
    {
        public static string AlignLine(string line, int width, string mode)
        {
            var visible = TextMeasure.VisibleLength(line);
            var slack = width - visible;
            if (slack <= 0)
            {
                return line;
            }
            if (mode == "right")
            {
                return new string(' ', slack) + line;
            }
            if (mode == "center")
            {
                var left = slack / 2;
                return new string(' ', left) + line + new string(' ', slack - left);
            }
            return line + new string(' ', slack);
        }
    }
}

File: fixtures/scale/swift/small/Render/Alignment.swift
import Foundation

enum Alignment {
    static func alignLine(_ line: String, width: Int, mode: String) -> String {
        let visible = TextMeasure.visibleLength(line)
        let slack = width - visible
        guard slack > 0 else { return line }
        if mode == "right" {
            return String(repeating: " ", count: slack) + line
        }
        if mode == "center" {
            let left = slack / 2
            return String(repeating: " ", count: left) + line + String(repeating: " ", count: slack - left)
        }
        return line + String(repeating: " ", count: slack)
    }
}
